setup.py

```diff
@@ -5,8 +5,33 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+import itertools
+
 from setuptools import setup
 
+EXTRAS = {
+    ':python_version=="2.7"': [
+        "backports.tempfile>=1.0",
+        "functools32>=3.2.3.post2",
+        "mercurial>=4.5.3",
+        "subprocess32>=3.5.0rc1",
+    ]
+}
+EXTRAS[""] = list(set(itertools.chain.from_iterable(EXTRAS.values())))
```

Comment:
What does this do?

Reply:
This was a way I could get the Python 2.7-only packages to work well with the whole new `.` and `.test` split.

Reply:
I did this in a cleaner way in rev b98f8ea83b617ab9c1b416d1871e1deca8e65aee.